rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp

```diff
@@ -200,3 +203,40 @@ TEST_F(TemporaryDirectoryFixture, can_write_mcap_with_zstd_configured_from_yaml)
   }
 }
 #endif  // #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
+
+TEST_F(TemporaryDirectoryFixture, mcap_contains_ros_distro)
+{
+  // const auto expected_file = rcpputils::fs::path(temporary_dir_path_) / "rosdistro_bag.mcap";
+  const auto expected_file = rcpputils::fs::path("rosdistro_bag.mcap");
+  const auto uri = rcpputils::fs::remove_extension(expected_file);
+  const std::string storage_id = "mcap";
+  const std::string current_ros_distro = rcpputils::get_env_var("ROS_DISTRO");
+  ASSERT_FALSE(current_ros_distro.empty());
+  std::string read_metadata_ros_distro = "";
+
+  // Open writer to create no-data file and then delete the writer to close
+  rosbag2_storage::StorageFactory factory;
+  rosbag2_storage::StorageOptions options;
+  options.uri = uri.string();
+  options.storage_id = storage_id;
+  auto writer = factory.open_read_write(options);
+  writer.reset();
+  ASSERT_TRUE(expected_file.is_regular_file());
+
+  // Open created mcap file, read all metadata records to find rosbag2.ROS_DISTRO value
+  mcap::Status status{};
```

Comment:
@emersonknapp I was expecting that you will add a new public method to the mcap_storage.cpp(hpp) aka `get_ros_distro_name()` for calling it from unit test. 
According to the https://www.acodersjourney.com/20-abi-breaking-changes/ 
> 3. Add new non-virtual methods to an existing class

These are ABI-compatible changes.

Reply:
There is no `.hpp` file - there's not really a point in adding such a new method since the only exposed API is via `ReadWriteInterface`

Reply:
I personally believe in the school of thought that we shouldn't have to add to feature code to serve unit tests. If the user-facing API is not testable, then it's maybe a bad API. In this case, we produce an mcap file as output and I'd rather just inspect that to see if it's correct than change the writer somehow to have it tell me that it's doing something correctly - that's testing the testing code, not testing the true outcome.